featurebyte/query_graph/sql/interpreter/preview.py

```diff
@@ -567,8 +579,34 @@ def _construct_count_stats_sql(
             expressions.Subquery(this=cat_count_dict, alias="count_dict")
         )
 
+    @staticmethod
+    def _get_ctes_with_casted_data(
+        sql_tree: expressions.Expression,
+    ) -> Tuple[expressions.Select, List[CteStatement]]:
+        cte_statements: List[CteStatement] = [("data", sql_tree)]
+
+        # get subquery with columns casted to string to compute value counts
+        casted_columns = []
+        for col_expr in sql_tree.expressions:
+            col_name = col_expr.alias or col_expr.name
+            # add casted columns
+            casted_columns.append(
+                expressions.alias_(
+                    expressions.Cast(this=quoted_identifier(col_name), to=parse_one("STRING")),
+                    col_name,
+                    quoted=True,
+                )
+            )
+        sql_tree = expressions.select(*casted_columns).from_("data")
+        cte_statements.append(("casted_data", sql_tree))
```

Comment:
seems like `"casted_data"` is a special string, so might be worth extracting to a constant too